packages/teleport/src/Console/Console.tsx

```diff
@@ -37,7 +38,8 @@ const POLL_INTERVAL = 5000; // every 5 sec
 
 export default function Console() {
   const consoleCtx = useConsoleContext();
-  const { verifyAndConfirm } = useOnExitConfirmation(consoleCtx);
+  const webSession = useWebSession();
```

Comment:
For consistency sake, it seems that for components with their own `use<component>Context` functions the session logout is handled there instead of in the component. See `useDiscover`, `useTopBar`, and `Player` below.

Reply:
I'd argue for using `useWebSession` _more_ in components than hiding it in these magic single hook methods we've created for the state management of entire components and their subtrees.

`useDiscover` shouldn't know anything about logging the user out - why would it? It just needs to know about discover and anything discover related. The component that actually displays the logout button should be the one calling `useWebSession`.

Our abstraction of everything into single hooks is confusing and bloated, and causes inefficient re-renders of our UI.

Reply:
I'm happy with what @ryanclark has said....my concern was just one of consistency.